app/modules/annotations/schemas.py

```diff
@@ -56,3 +56,80 @@ class Meta(BaseAnnotationSchema.Meta):
             Annotation.updated.key,
             'asset_src',
         )
+
+
+def get_locationId(annot):
+    enc, enc_edm, sight_edm = annot.get_related_extended_data()
+    if not enc or not enc_edm or not sight_edm:
+        return None
+    return enc_edm.get('locationId') or sight_edm.get('locationId')
+
+
+def get_taxonomy_guid(annot):
+    return annot.get_taxonomy_guid(sighting_fallback=True)
+
+
+def get_time(annot):
+    enc, enc_edm, sight_edm = annot.get_related_extended_data()
+    if not enc:
+        return None
+    return enc.get_time_isoformat_in_timezone(sighting_fallback=True)
+
+
+def get_owner_guid(annot):
+    guid = annot.get_owner_guid()
+    return str(guid) if guid else None
+
+
+def get_encounter_guid(annot):
+    return str(annot.encounter_guid) if annot.encounter_guid else None
+
+
+def get_sighting_guid(annot):
+    return (
+        str(annot.encounter.sighting.guid)
+        if annot.encounter and annot.encounter.sighting
+        else None
+    )
+
+
+def get_keywords_flat(annot):
+    if not annot.keyword_refs:
+        return []
+    return sorted([ref.keyword.value for ref in annot.keyword_refs])
+
+
+class AnnotationElasticsearchSchema(BaseAnnotationSchema):
+    """
+    Schema for indexing by Elasticsearch
+
+    Note: can be expensive (as it delves into related objects as well as EDM), so best not to use
+    for purposes other than ES indexing.
+    """
+
+    keywords = base_fields.Function(get_keywords_flat)
+    locationId = base_fields.Function(get_locationId)
+    taxonomy_guid = base_fields.Function(get_taxonomy_guid)
+    owner_guid = base_fields.Function(get_owner_guid)
```

Comment:
Why have owner guid? All other APIs have the owner PublicUserSchema.

Reply:
this schema is for **elasticsearch indexing** only -- so it is what we want to search against -- not something that is used like a normal api. e.g. one would want to simply match the owner _guid_, not the details about that User.   similarly, i made the **keywords** just the list of text-values basically -- with the assumption that would be enough to search against (via Elasticsearch).  however, i could see modifying this one to _also_ allow searching on the keyword guid as well.

in short, these elastic-search-specific schema are what the ES _document_ ends up looking like.

Reply:
addressing in refactored code